fault/fault_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fault
+
+import (
+	"testing"
+
+	typev3 "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	"github.com/gogo/googleapis/google/rpc"
+)
+
+func TestErrorMessage(t *testing.T) {
+	tests := []struct {
+		desc                 string
+		fault                *AdapterFault
+		expectedErrorMessage string
+	}{
+		{
+			desc:                 "test0",
+			fault:                CreateAdapterFault("test-code", rpc.DEADLINE_EXCEEDED, typev3.StatusCode_Forbidden),
+			expectedErrorMessage: "FaultCode:test-code, RpcCode:DEADLINE_EXCEEDED, StatusCode:Forbidden",
+		},
+		{
+			desc:                 "test1",
+			fault:                CreateAdapterFault("", 1, 0),
+			expectedErrorMessage: "FaultCode:, RpcCode:CANCELLED, StatusCode:Empty",
+		},
+		{
+			desc:                 "test2",
+			fault:                nil,
+			expectedErrorMessage: "AdapterFault: Error() called on a nil object",
+		},
+	}
+
+	for _, test := range tests {
+		t.Logf("Running test %v", test.desc)
```

Comment:
This is not needed. And if you are iterating over tests,  the standard pattern in the loop should be:
```
t.Run(test.Desc, func(t *testing.T) {
    // your existing testing logic
})
```

Reply:
Thanks for pointing out, Done.